src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1133,6 +1134,8 @@ RemoveNodeFromCluster(char *nodeName, int32 nodePort)
 		}
 	}
 
+	DeleteWorkerTransactions(workerNode);
```

Comment:
Citus could have two node roles for the same groupId:
```SQL
select groupid, nodename,nodeport,noderole from pg_dist_node;
 groupid |   nodename    | nodeport | noderole  
---------+---------------+----------+-----------
       1 | 192.168.1.110 |    10004 | primary
       1 | 192.168.1.110 |    10005 | secondary
(2 rows)

```

And, as far as I know, it is strictly forbidden to have  a 2PC on the  secondary nodes. Secondary nodes are read-only,  hence never 2PC is used. Hence, no items can be inserted to pg_dist_transaction for secondary nodes.

If the `master_remove_node` is on the secondary node, we should not be calling `DeleteWorkerTransactions ()` as it could break on going 2PCs on its primary. 

So, the `DeleetWorkerTransactions()` should be called inside `if (NodeIsPrimary(workerNode))`

See the related regression tests and code: https://github.com/citusdata/citus/blob/ba300dcad8f7ff6cbbfbe8c167de0109d32ee9c6/src/backend/distributed/metadata/metadata_cache.c#L280-L293